packages/flutter/lib/src/widgets/image_filter.dart

```diff
@@ -46,32 +65,75 @@ class ImageFiltered extends SingleChildRenderObjectWidget {
   }
 }
 
-class _ImageFilterRenderObject extends RenderProxyBox {
-  _ImageFilterRenderObject(this._imageFilter);
+/// Applies an [ImageFilter] to its child.
+class RenderImageFiltered extends RenderProxyBox {
+  /// Creates a [RenderObject] that applies an [ImageFilter] to its child.
+  ///
+  /// One of the [imageFilter] or [imageFilterCallback] should not be null here
+  /// or whenever the [paint] method is invoked.
+  /// The [imageFilter] will be used if it is not null, otherwise the filter
+  /// will be obtained from the [imageFilterCallback].
+  RenderImageFiltered({
+    required ImageFilter? imageFilter,
+    required ImageFilterCallback? imageFilterCallback,
+  })
+      : assert(imageFilter != null || imageFilterCallback != null),
```

Comment:
your asserts aren't as strict here as in the widget, is that intentional?

Reply:
The basic problem I was facing was that the setters for these properties would be applied independently. So, if you switch from a widget that had a filter to one that had a callback, then when the RO was updated there would be a brief moment when both were null (or vice versa if you switched the other way).

Since the RO was not going to always have both conditions, I decided to relax its assertions. In fact, it only really matters that at least one is provided, but providing both likely indicates sloppy code or sloppy understanding of the object.

In retrospect, no matter what conditions I assert in the constructor and paint, there will always be a moment in time between them when the conditions could be violated by the individual setters, so I might as well assert the same conditions as the widget and just document that they only need to apply when constructing and when paint is called. Other ROs document a condition that must be true by the time the paint or compositing steps happen.

Reply:
I think I've sold myself on using the same conditions as widget and just stating that they are only enforced in constructor and paint...

Reply:
In the latest commit, the ImageFiltered constructor, the RenderImageFiltered constructor, and RenderImageFiltered paint method all now enforce the consistent "one or the other, but not both" condition.